lib/streamlit/util.py

```diff
@@ -212,3 +212,21 @@ def extract_key_query_params(
             for item in sublist
         ]
     )
+
+
+def extract_single_element_lists(
+    dict: Dict[str, List[Any]]
+) -> Dict[str, Union[Any, List[Any]]]:
+    """
+    Extracts the single element from lists in a dictionary if the list has a length of 1.
+
+    Args:
+    - dict (dict): The dictionary containing lists.
+
+    Returns:
+    - dict: The modified dictionary with single element lists extracted.
```

Comment:
This doesn't follow [numpydoc](https://numpydoc.readthedocs.io/en/latest/format.html) style like most of our docstrings do. See the docstring for the `index_` function in this same file for an example